packages/ember-toucan-core/src/components/form/controls/combobox.ts

```diff
@@ -0,0 +1,537 @@
+  @tracked inputValue: string | undefined;
+  @tracked isPopoverOpen = false;
+  @tracked filteredOptions: OPTION[] | undefined;
+
+  Chevron = Chevron;
+  Option = OptionComponent;
+  popoverId = `popover--${guidFor(this)}`;
+
+  constructor(
+    owner: unknown,
+    args: ToucanFormComboboxControlComponentSignature<OPTION>['Args']
+  ) {
+    super(owner, args);
+
+    // We need to set our input tag's value attribute
+    // if we have a selected option provided on render
+    let { selected, optionKey } = this.args;
+
+    this.inputValue =
+      typeof selected === 'object' && optionKey
+        ? (selected[optionKey] as string | undefined)
+        : (selected as string | undefined);
+  }
+
+  velcroMiddleware: VelcroMiddleware[] = [
+    offset({
+      mainAxis: 8,
+    }),
+    size({
+      apply({ rects, elements }) {
+        Object.assign(elements.floating.style, {
+          width: `${rects.reference.width}px`,
+        });
+      },
+    }),
+  ];
+
+  /**
+   * This is required for accessibility so that we can announce to the screenreader the highlighted option as the user uses the arrow keys.
+   */
+  get activeDescendant() {
+    // For the case where nothing is selected on render
+    if (this.activeIndex === null) {
+      return null;
+    }
+
+    return `${this.popoverId}-${this.activeIndex}`;
+  }
+
+  /**
+   * This state is used to determine if we should add event handlers to the input element or not.
+   */
+  get isDisabledOrReadOnlyOrWithoutOptions() {
+    return (
+      this.args.isDisabled ||
+      this.args.isReadOnly ||
+      this.args.options === undefined
+    );
+  }
+
+  /**
+   * We apply different styles to the input tag based on our current state.
+   */
+  get styles() {
+    if (this.args.isDisabled) {
+      return 'shadow-focusable-outline bg-overlay-1 text-disabled pointer-events-none placeholder:text-disabled';
+    }
+
+    if (this.args.isReadOnly) {
+      return 'focus:shadow-focus-outline bg-surface-xl shadow-read-only-outline text-titles-and-attributes placeholder:text-titles-and-attributes';
+    }
+
+    if (this.args.hasError) {
+      return 'shadow-error-outline focus:shadow-error-focus-outline bg-overlay-1 text-titles-and-attributes';
+    }
+
+    return 'shadow-focusable-outline focus:shadow-focus-outline bg-overlay-1 text-titles-and-attributes';
+  }
+
+  /**
+   * The options to render inside of the popover list.
+   */
+  get options() {
+    return this.filteredOptions || this.args?.options;
+  }
+
+  /**
+   * The internal currently selected item.
+   */
+  get selected(): string | undefined {
+    let { optionKey, selected } = this.args;
+
+    if (!selected) {
+      return undefined;
+    }
+
+    return (
+      typeof selected === 'object' && optionKey ? selected[optionKey] : selected
+    ) as string;
+  }
+
+  /**
+   * Attempts to scroll the active or newly highlighted item into view for the user.
+   */
+  #scrollActiveOptionIntoView(alignToTop?: boolean) {
```

Comment:
Why a private member here? I mean it's not wrong, but the (Glimmer) component model is basically that _all_ props and methods are private effectively, as you can only pass args from the outside. So seems inconsistent, and not really needed here...

Reply:
Removed as part of https://github.com/CrowdStrike/ember-toucan-core/pull/200/commits/754b29b79b808728f95bcc2885851e362839c53b!

Reply:
Just came across this! 

It was my doing. I tend to make private things private regardless of how the framework treats them. But, even though Glimmer handles privacy for us, I still think private methods are useful in that they imply a method is not used in the template.

Or is the fact that the a private method can't be used in the template just a shortcoming of the template syntax and not by design?